File: frontend/bff/src/application/organizations/commands/add-commodity.command.ts
import type { Commodity } from '../../../cross-cutting/enums/commodity.js';
import type { MeasurementUnit } from '../../../cross-cutting/enums/measurement-unit.js';
import type { RequestContext } from '../../../cross-cutting/context/request-context.js';

export interface AddCommodityCommand {
  context: RequestContext;
  commodity: Commodity;
  capacity: number;
  unit: MeasurementUnit;
  priceReference: string | null;
  currency: string;
  sells: boolean;
}

File: backend/src/Fix.Application/UseCases/Organizations/Commands/AddCommodityCommand.cs
using Fix.Application.Abstractions.Authorization;
using Fix.Application.Abstractions.Context;
using Fix.Application.Abstractions.Messaging;
using Fix.Application.Organizations.Dtos;
using Fix.Domain.Common;
using Fix.Domain.AggregateRoots.Roles;

namespace Fix.Application.Organizations.Commands;

[RequireRole(RoleCodes.EditOrganization)]
public sealed record AddCommodityCommand(
    Guid UserId,
    Guid OrganizationId,
    Commodity Commodity,
    decimal Capacity,
    MeasurementUnit Unit,
    string? PriceReference,
    string Currency,
    bool Sells)
    : ICommand<OrganizationSetupDto>, IOrganizationRequest;

